portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,21 +65,48 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     } catch (UnsupportedEncodingException e) {
       // Send a HTTP 400 Bad Request response if user provided invalid data.
       response.sendError(HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
-    } 
-    catch (Exception e) {
+    } catch (Exception e) {
       // Send a HTTP 500 error for other exceptions
       response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getMessage());
     }
-    
-    comments.add(newComment);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(getCommentEntity(newComment));
 
     // Redirect back to the HTML page.
     response.sendRedirect("/index.html");
   }
 
+  private List<String> getCommentsArray(int limit) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    List<String> comments = new ArrayList<>();
+    for (Entity comment: results.asIterable()) {
+      comments.add((String)comment.getProperty("message"));
+
+      -- limit;
+      if (limit == 0) {
+        break;
+      }
+    }
+    return comments;
+  }
+
+  /** Creates Entity with a kind of Comment */
+  private Entity getCommentEntity(String newComment) {
```

Comment:
nit: createCommentEntity or buildCommentEntity might be a better name for this helper.